### PR TITLE
Make Linux Great Again

### PR DESCRIPTION
**Pull request by OmniTroid**
should fix the issues with missing files in Linux. some refactoring and removing unused legacy support as well.

### CODE REVIEW

File: path_functions.cpp [hunk 1]
```diff
@@ -4,104 +4,159 @@
...
-QString Courtroom::get_default_background_path()
-{
-  return ao_app->get_base_path() + "background/default/";
+  //quick check to see if it's actually there first
+  if (file.exists()) return p_file;
+
+  QString file_name = file.fileName();
+  QString file_path = file.absolutePath();
+
+  QRegExp file_rx = QRegExp(file_name, Qt::CaseInsensitive);
```
Comment (oldmud0) [inline path_functions.cpp]: Does this work if the directory is case-insensitive as well?

Comment (OmniTroid) [reply path_functions.cpp]: it assumes that the parent of p_file definitely exists (with proper case) and does not differentiate between directory and file. so for example, for the file:

base/characters/SoMeThInG/emotions/BUTTON1_OFF.PNG

this need to be called twice; the first call would be get_case_sensitive_path("base/characters/SoMeThInG") -> return "base/characters/Something"(the file that actually exists), and the second call would be
get_case_sensitive_path("base/characters/Something/emotions/BUTTON1_OFF.PNG") -> return "base/characters/Something/emotions/button1_off.png"(again, a file that actually exists)

the solution is sound and works on Ubuntu 18.10 with ext4 filesystem. ive yet to test it with cyrillic characters, but assuming Qt is smart with encodings it should work

**Comment by OmniTroid** [comment]
also tested with cyrillic script, works fine

File: path_functions.cpp (hunk 1 above)
Comment (oldmud0) [reply path_functions.cpp]: so, it's not recursive by default? the issue is that some asset authors have taken it upon themselves to organize their emotes by hacking their own directories into char.ini files. Those should be checked as well.

Comment (OmniTroid) [reply path_functions.cpp]: nope, I did not thoroughly consider that case (honestly hoped it had fallen out of fashion by now). I suppose it's not too much of a performance issue to check for existence of parent folder though. ill make it recursive

**Comment by OmniTroid** [comment]
Made get_case_sensitive path recursive. tested as well

**Comment by OmniTroid** [comment]
so just close this PR then?

**Comment by oldmud0** [comment]
uh, yeah, it'll close itself once mega-merge is merged in

**Pull request merged**